File: warehouse-management-web-angular/src/app/warehouse/document-receipt/models/input/create-resource-receipt-input.ts
import { IncludeResourceReceiptInput } from './include-resource-receipt-input';

/**
 *  Класс входной модели создания ресурса поступления.
 */
export class CreateResourceReceiptInput {
  /**
   * Номер документа поступления.
   */
  documentReceiptNumberCode: string = '';
  /**
   * Дата документа поступления.
   */
  date: Date = new Date();
  /**
   * Id клиента документа поступления.
   */
  clientId: number = 0;
  /**
   * Список входящих ресурсов поступления.
   */
  includeResourceReceiptInputs: IncludeResourceReceiptInput[] | null = null;
}

File: WarehouseManagementWeb.Application/Dto/Input/ResourceReceipt/CreateResourceReceiptInput.cs
namespace WarehouseManagementWeb.Application.Dto.Input.ResourceReceipt
{
    /// <summary>
    /// Класс входной модели создания ресурса поступления.
    /// </summary>
    public class CreateResourceReceiptInput
    {
        /// <summary>
        /// Номер документа поступления. 
        /// </summary>
        public string? DocumentReceiptNumberCode { get; set; }

        /// <summary>
        /// Дата документа поступления.
        /// </summary>
        public DateTime Date { get; set; }

        /// <summary>
        /// Id клиента документа поступления.
        /// </summary>
        public int ClientId { get; set; }

        /// <summary>
        /// Список входящих ресурсов в документ поступления.
        /// </summary>
        public ICollection<IncludeResourceReceiptInput>? IncludeResourceReceiptInputs { get; set; }
    }
}
